app/script/calling/entities/FlowEntity.js

```diff
@@ -1222,25 +1226,77 @@ z.calling.entities.FlowEntity = class FlowEntity {
   //##############################################################################
 
   /**
-   * Update the local MediaStream.
+   * Replace the MediaStream attached to the PeerConnection.
+   *
+   * @private
    * @param {z.media.MediaStreamInfo} mediaStreamInfo - Object containing the required MediaStream information
-   * @returns {Promise} Resolves when the updated MediaStream is used
+   * @param {MediaStream} outdatedMediaStream - Previous MediaStream
+   * @returns {Promise} Resolves when negotiation has been restarted
    */
-  updateMediaStream(mediaStreamInfo) {
-    return this._replaceMediaTrack(mediaStreamInfo).catch(error => {
-      const {message, type} = error;
-      const expectedErrorTypes = [
-        z.calling.CallError.TYPE.NO_REPLACEABLE_TRACK,
-        z.calling.CallError.TYPE.RTP_SENDER_NOT_SUPPORTED,
-      ];
+  replaceMediaStream(mediaStreamInfo, outdatedMediaStream) {
+    return Promise.resolve()
+      .then(() => this._removeMediaStream(outdatedMediaStream))
+      .then(() => {
+        const {stream: mediaStream} = mediaStreamInfo;
+        this.restartNegotiation(z.calling.enum.SDP_NEGOTIATION_MODE.STREAM_CHANGE, false, mediaStream);
+      })
+      .catch(error => {
+        this.callLogger.error(`Failed to replace local MediaStream: ${error.message}`, error);
+        throw error;
+      });
+  }
 
-      const isExpectedError = expectedErrorTypes.includes(type);
-      if (isExpectedError) {
-        this.callLogger.debug(`Replacement of MediaStream and renegotiation necessary: ${message}`, error);
-        return this._replaceMediaStream(mediaStreamInfo);
-      }
-      throw error;
-    });
+  /**
+   * Replace the MediaStreamTrack attached to the MediaStream of the PeerConnection.
+   *
+   * @private
+   * @param {z.media.MediaStreamInfo} mediaStreamInfo - Object containing the required MediaStream information
+   * @returns {Promise} Resolves when a MediaStreamTrack has been replaced
+   */
+  replaceMediaTrack(mediaStreamInfo) {
+    const {stream: mediaStream, type: mediaType} = mediaStreamInfo;
+    const [mediaStreamTrack] = mediaStream.getTracks();
```

Comment:
Is it always safe to assume that we just need the first track? Are we absolutely sure we never need to handle multiple tracks?

Reply:
Yes, in this case it is.